postgres_insert_block.py

```diff
@@ -0,0 +1,62 @@
+from nio.properties import BoolProperty, StringProperty
+from nio.util.discovery import discoverable
+from .postgres_base_block import PostgresBase
+
+
+@discoverable
+class PostgresInsert(PostgresBase):
+    """A block for inserting incoming signals into a postgres database.
+
+    Properties:
+        table_name(str): name of the table on the database to execute commands
+                         on.
+
+    """
+
+    upsert = BoolProperty(title="Update conflicting keys", default=True)
+    table_name = StringProperty(title="Table name", allow_none=False)
+
+    def __init__(self):
+        super().__init__()
+
+    def process_signals(self, signals):
+        """execute an insert command for all incoming signals"""
+        for signal in signals:
+            self.execute_insert(signal.to_dict())
+
+    def execute_insert(self, data):
+        """execute an insert query for the given data"""
+        self.logger.debug('executing INSERT on data: {}'.format(data))
+        try:
+            self._cur.execute(self._build_insert_query_string(data),
+                              tuple(data.values()))
+        except:
+            self.logger.exception("Could not execute command".format())
+            self._rollback_transactions()
+        else:
+            self._commit_transactions()
+
+    def _build_insert_query_string(self, data):
+        """build an INSERT SQL query based on the incoming (dictionary)
+        data.
+        """
+        query_base = 'INSERT INTO {} ({}) VALUES ({})'
+        # TODO: possible sql injection on db name and data keys?
+        query_final = query_base.format(self.table_name(),
+                                        ', '.join(list(data.keys())),
```

Comment:
Maybe out of scope for this initial PR, but I think eventually we should be able to set the columns we want to insert in the block's configuration. Doing this type of dynamic column construction doesn't jive well with traditional SQL databases like Postgres.